pkg/controller/shared_state.go

```diff
@@ -86,8 +88,30 @@ type SharedState struct {
 	azureDiskCSITranslator        csitranslator.InTreePlugin
 	availableAttachmentsMap       sync.Map
 	driverLifecycle               DriverLifecycle
+	eventsToPersistQueue          chan ReplicaAttachmentFailureInfo
+	eventsToUnpersistQueue        chan string
+	_eventRefresherInitOnce       sync.Once
 }
 
+type ReplicaAttachmentFailureInfo struct {
+	volumeName string
+	message    string
+	pods       []runtime.Object
+	timestamp  time.Time
+}
+
+const (
+	// persistent event and their refreshes should overlap by this amount of time
+	eventOverlapDuration = time.Millisecond
+	// allow persistent events to overlap by up to this additional amount of time for batched reporting etc.
+	eventOverlapVariance = time.Millisecond
```

Comment:
I believe 1ms would be too small for actually end up batching. Since batching would result in more calls to API server in a short window of time thereby increasing the likelihood of throttling, should we even bother?